E0528 needs to be updated to new format

From: src/test/compile-fail/E0528.rs

E0528 needs a span_label, updating it from:

```
error[E0528]: pattern requires at least 3 elements but array has 2
  --> src/test/compile-fail/E0528.rs:16:10
   |
16 |         &[a, b, c, rest..] => { //~ ERROR E0528
   |          ^^^^^^^^^^^^^^^^^
```

To:

```
error[E0528]: pattern requires at least 3 elements but array has 2
  --> src/test/compile-fail/E0528.rs:16:10
   |
16 |         &[a, b, c, rest..] => { //~ ERROR E0528
   |          ^^^^^^^^^^^^^^^^^ pattern cannot match array of 2 elements
```

--- a/src/librustc_typeck/check/_match.rs
+++ b/src/librustc_typeck/check/_match.rs
@@ -248,9 +248,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                         } else if let Some(rest) = size.checked_sub(min_len) {
                             (inner_ty, tcx.mk_array(inner_ty, rest))
                         } else {
-                            span_err!(tcx.sess, pat.span, E0528,
-                                      "pattern requires at least {} elements but array has {}",
-                                      min_len, size);
+                            struct_span_err!(tcx.sess, pat.span, E0528,
+                                    "pattern requires at least {} elements but array has {}",
+                                    min_len, size)
+                                .span_label(pat.span,
+                                    &format!("pattern cannot match array of {} elements", size))
+                                .emit();
                             (inner_ty, tcx.types.err)
                         }
                     }

--- a/src/test/compile-fail/E0528.rs
+++ b/src/test/compile-fail/E0528.rs
@@ -13,7 +13,9 @@
 fn main() {
     let r = &[1, 2];
     match r {
-        &[a, b, c, rest..] => { //~ ERROR E0528
+        &[a, b, c, rest..] => {
+        //~^ ERROR E0528
+        //~| NOTE pattern cannot match array of 2 elements
         }
     }
 }

